compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7549,7 +7549,8 @@ private BType checkArrayIndexBasedAccess(BLangIndexBasedAccess indexBasedAccess,
                     actualType = arrayType.eType;
                     break;
                 }
-                actualType = getConstIndex(indexExpr) >= arrayType.size ? symTable.semanticError : arrayType.eType;
+                var indexVal = getConstIndex(indexExpr);
```

Comment:
From a readability perspective, should we avoid using `var`?

Reply:
yes, we need to avoid using var in java code.

Reply:
fixed